flink-clients/src/main/java/org/apache/flink/client/deployment/application/ClassPathPackagedProgramRetriever.java

```diff
@@ -111,6 +112,20 @@ private ClassPathPackagedProgramRetriever(
 	@Override
 	public PackagedProgram getPackagedProgram() throws FlinkException {
 		try {
+
+			// It is Python job if program arguments contain "-py"/--python" or "-pym/--pyModule", set the fixed
+			// jobClassName and jarFile path.
+			if (PackagedProgramUtils.isPython(jobClassName) || PackagedProgramUtils.isPython(programArguments)){
+				String pythonJobClassName = PackagedProgramUtils.PYTHON_DRIVER_CLASS_NAME;
+				File pythonJarFile = new File(PackagedProgramUtils.getPythonJar().getPath());
```

Comment:
Could we get a correct `pythonJarFile` for Yarn perjob/application mode here? It seems that we do not ship the $FLINK_HOME/opt directory and set the `FLINK_OPT_DIR` environment.

Maybe it is out of the scope of this PR.

Reply:
Currently, we have no plan to support running Pyflink on yarn in application mode.